provider/terraform.rb

```diff
@@ -67,19 +67,12 @@ def force_new?(property, resource)
                              force_new?(property.parent, resource))))
     end
 
-    # Puts together the links to use to make API calls for a given resource type
-    def self_link_url(resource)
-      (product_url, resource_url) = self_link_raw_url(resource)
-      [product_url, resource_url].flatten.join
-    end
-
-    def collection_url(resource)
-      base_url = resource.base_url.split("\n").map(&:strip).compact
-      [resource.__product.base_url, base_url].flatten.join
+    def build_url(url_parts, _extra = false)
+      url_parts.flatten.join
     end
 
     def update_url(resource, url_part)
-      return self_link_url(resource) if url_part.nil?
+      return build_url(resource.self_link_url) if url_part.nil?
       [resource.__product.base_url, url_part].flatten.join
```

Comment:
Should this be `build_url([resource.__product.base_url, url_part])`? Or move this up into resource.rb with the other url methods?

Reply:
This seemed very Terraform exclusive for the time being. I didn't want to move it into api/resource.rb until it was more obvious that it would be used in the other three providers.

(Side note - I've only got a rough idea what this method does. What's url_part in this context?)

Reply:
Ah yeah, it's for constructing custom update urls. It's tf-specific right now just because only terraform has that functionality, but the others will need it eventually. url_part is the `update_url` from api.yaml. From a quick look we might actually be able to just get rid of that second parameter and use `resource.update_url` but I haven't tested it out.